### PR TITLE
docs/library/pyb.Timer.rst: Add `brk` argument to `init` method.

### PR DESCRIPTION
**Pull request by hlovatt**

### CODE REVIEW

File: docs/library/pyb.Timer.rst [hunk 1]
```diff
@@ -62,7 +62,7 @@ Constructors
 Methods
 -------
 
-.. method:: Timer.init(*, freq, prescaler, period, mode=Timer.UP, div=1, callback=None, deadtime=0)
+.. method:: Timer.init(*, freq, prescaler, period, mode=Timer.UP, div=1, callback=None, deadtime=0, brk)
```
Comment (dpgeorge) [inline docs/library/pyb.Timer.rst:65]: the default needs to be shown, ie `brk=BRK_OFF`

File: docs/library/pyb.Timer.rst [hunk 2]
```diff
@@ -107,6 +107,12 @@ Methods
        measures ticks of ``source_freq`` divided by ``div`` clock ticks.
        ``deadtime`` is only available on timers 1 and 8.
 
+     - ``brk`` - specifies if the break mode is used to kill the output of
+        the PWM when the ``BRK_IN`` input is asserted. The polarity set how the
```
Comment (dpgeorge) [inline docs/library/pyb.Timer.rst:111]: would read better something like: "... input is asserted.  The value of this argument determines if break is enabled and what the polarity is, and can be one of ``BRK_OFF``, ``BRK_LOW`` or ``BRK_HIGH``.

I understand you copied this from the code, but the docs should be the source of truth, and the comments in the code eventually removed.

**Comment by dpgeorge** [comment]
Updated as suggested in the comments above and merged in 89e1e67748e681da5f414dc597e89eaee8822817

**Comment by chrismas9** [comment]
It's not clear from the docs how to select the brk input.  On F405 TIM1 has three BKIN pins (PA6, PB12 and PE15). On newer parts there is also BKIN2. brk won't work unless the pin af is set to forward a pin input to the timer block.
Since Timer.init doesn't allow you to specifiy the BKIN pin I tried setting the Pin alt value. The pyb.Pin class appears to only allow alt values for outputs:

```
MicroPython v1.19.1-19-g6d53022ae-dirty on 2022-06-21; NUCLEO-L496ZG-P with STM32L496

Type "help()" for more information.
>>> import pyb
>>> from pyb import Pin
>>> mypin =pyb.Pin.cpu.A6
>>> mypin.af_list()
[Pin.AF1_TIM1, Pin.AF2_TIM3, Pin.AF3_TIM8, Pin.AF5_SPI1, Pin.AF7_USART3, Pin.AF8_LPUART1, Pin.AF14_TIM16]
>>> mypin.af()
0
>>> mypin.init(mode=Pin.AF_PP, alt=Pin.AF2_TIM3)
>>> mypin.af()
2
>>> mypin.init(mode=Pin.IN, alt=Pin.AF1_TIM1)
>>> mypin.af()
2
>>> mypin.init(mode=Pin.IN)
>>> mypin.af()
2
>>> mypin.init(mode=Pin.OUT_PP)
>>> mypin.af()
2
>>> 

```

As you can see setting an output af works, but setting an input af leaves the alt value unchanged. On PA6 `Pin.AF1_TIM1` is `TIM1_BKIN`. I also noticed that just setting a pin back to a standard input or output doesn't reset the af mode.

Maybe there is a way, but I can't find it from my reading of the docs. It would be nice if you could specify a Pin object for the brk input in `Timer.init` in the same way you specify a pin output for PWM in the TimerChannel class.

My questions are:

1. Does brk in work with an external pin?
2. How do you configure it?
3. If no, how to fix it - in Pin or Timer (preference Timer)? I don't think  can help with C code.
4. If yes have I missed something in ther docs? I can update the docs if I knew how it works.

**Comment by chrismas9** [comment]
I have resolved my queries. See
[https://github.com/micropython/micropython/pull/8823](url)

**Pull request closed without merging**